consensus/src/round_manager.rs

```diff
@@ -920,6 +942,69 @@ impl RoundManager {
         Ok(vote)
     }
 
+    async fn process_order_vote_msg(&mut self, order_vote: OrderVote) -> anyhow::Result<()> {
+        if self.onchain_config.order_vote_enabled() {
+            fail_point!("consensus::process_order_vote_msg", |_| {
+                Err(anyhow::anyhow!("Injected error in process_order_vote_msg"))
+            });
+            info!(self.new_log(LogEvent::ReceiveOrderVote), "{}", order_vote);
+
+            if self
+                .pending_order_votes
+                .has_enough_order_votes(order_vote.ledger_info())
+            {
+                return Ok(());
+            }
+
+            if order_vote.ledger_info().round()
+                > self.block_store.sync_info().highest_commit_round()
+            {
+                let vote_reception_result = self
+                    .pending_order_votes
+                    .insert_order_vote(&order_vote, &self.epoch_state.verifier);
+                self.process_order_vote_reception_result(&order_vote, vote_reception_result)
+                    .await?;
+            } else {
+                ORDER_VOTE_VERY_OLD.inc();
+                info!(
+                    "Received old order vote. Order vote round: {:?}, Highest commit round: {:?}",
+                    order_vote.ledger_info().round(),
+                    self.block_store.sync_info().highest_commit_round()
+                );
+            }
+        }
+        Ok(())
+    }
+
+    async fn broadcast_order_vote(
+        &mut self,
+        vote: &Vote,
+        qc: Arc<QuorumCert>,
+    ) -> anyhow::Result<()> {
+        if let Some(proposed_block) = self.block_store.get_block(vote.vote_data().proposed().id()) {
+            // Generate an order vote with ledger_info = proposed_block
+            let order_vote_proposal = proposed_block.order_vote_proposal(qc);
+            let order_vote_result = self
+                .safety_rules
+                .lock()
+                .construct_and_sign_order_vote(&order_vote_proposal);
+            let order_vote = order_vote_result.context(format!(
+                "[RoundManager] SafetyRules Rejected {} for order vote",
+                proposed_block.block()
+            ))?;
+            if !proposed_block.block().is_nil_block() {
+                observe_block(
+                    proposed_block.block().timestamp_usecs(),
+                    BlockStage::ORDER_VOTED,
```

Comment:
i think this is duplicate of QC aggregated?

Reply:
Should be pretty close. The difference would be the time taken to create an order vote.